content/6_Advanced/Treaps.mdx

```diff
@@ -326,6 +328,215 @@ With split and merge, how do we implement the additional operations?
   in lazy propogation). When splitting or merging, we push these tags downwards
   and perform the corresponding operation.
 
+### Implementation
+
+This code solves the above problem, but also provides a generalizable template.
+
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+// https://judge.yosupo.jp/problem/dynamic_sequence_range_affine_range_sum
+#include <bits/stdc++.h>
+#define int long long
+using namespace std;
+
+// BeginCodeSnip{ModInt class}
+// operations under mod
+struct mint;
+using P = pair<mint, mint>;
+struct mint {
```

Comment:
why  not declare the struct first and then `using P`
also maybe use uh smth like `pmint` instead